src/store/modules/emails.js

```diff
@@ -11,18 +10,16 @@ const getters = {
     //emails will be set within state from emails service
     return state.mailObject;
   },
-  getSelectedSenderObject(state, getters) {
+  getMailObjectById: (state, getters) => (id) => {
     //finds the emails attached to the selected sender's ID
-    return getters.getMailObject.find(
-      (sender) => sender.senderId === state.selectedSenderId
-    );
+    return getters.getMailObject.find((sender) => sender.id === id);
   },
   //returns all the emails, with most recent ones first
-  getEmailsOrderByDateAsc(state, getters) {
+  getMailObjectEmailsOrderByDateAsc: (state, getters) => (id) => {
     //get list of emails
-    const mail = getters.getSelectedSenderObject.emails;
+    var mailObject = { ...getters.getMailObjectById(id) };
```

Comment:
Ewww a `var`! Get it out ! If only a `let` was here to save us!

Reply:
It's gone... Pleassseee forgive me,I learned js so long ago